test/cartesian_range.jl

```diff
@@ -0,0 +1,20 @@
+using HDF5
+using Base.Test
+using Compat
+
+@testset "cartesian_range" begin 
+
+filename = "$(tempname()).h5"
+
+h5write(filename, "main", Array(1:10))
+@test h5read(filename, "main", CartesianRange((3:5,))) == [3:5;]
+f = h5open(filename,"r+")
+dset = f["main"]
+# setindex of cartesian range
+dset[CartesianRange((3:5,))] = [4:6;]
+@test dset[CartesianRange((3:5,))] == [4:6;]
```

Comment:
The tests should cover a case with more than one dimension. Also, why is the correct answer not `[3:5;]` as you would get with a normal range?

Reply:
because on  line 14 he rewrites elements 3:5 to 4:6 (which is testing setindex!)